virtualizarr/manifests/manifest.py

```diff
@@ -42,11 +46,20 @@ def to_kerchunk(self) -> List[Union[str, int]]:
         return [self.path, self.offset, self.length]
 
 
+# TODO we want the path field to contain a variable-length string, but that's not available until numpy 2.0
+# See https://numpy.org/neps/nep-0055-string_dtype.html
+MANIFEST_STRUCTURED_ARRAY_DTYPES = np.dtype(
+    [("path", "<U32"), ("offset", np.int32), ("length", np.int32)]
+)
```

Comment:
Because file paths can be strings of any length, we really need to be using numpy's new [variable-width string dtype](https://numpy.org/neps/nep-0055-string_dtype.html) here.

Unfortunately it's only coming out with numpy 2.0, and although there is a release candidate for numpy 2.0, it's so new that [pandas doesn't support it yet](https://github.com/pandas-dev/pandas/issues/55519). Xarray has a pandas dependency, so currently we can't actually build an environment that let's us try virtualizarr with the variable-length string dtype yet.

Reply:
Pandas just released [2.2.2](https://pandas.pydata.org/docs/whatsnew/v2.2.2.html), which is compatible with the upcoming numpy 2.0 release.

Not sure if that will break any part of xarray that we need for VirtualiZarr, but this might now be close enough to test out variable-length dtypes now.